### PR TITLE
Remove the `vendored` feature of `openssl-sys`

### PR DESCRIPTION
**Pull request by stormshield-kg**
Implements first part of https://github.com/sgrif/pq-sys/issues/58.

### CODE REVIEW

File: pq-src/Cargo.toml [hunk 1]
```diff
@@ -22,17 +22,13 @@ keywords = ["bindings", "libpq", "pq-sys", "bundled"]
...
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [target.'cfg(not(target_os = "windows"))'.dependencies]
-openssl-sys = { version = "0.9.93", features = ["vendored"] }
+openssl-sys = "0.9.93"
 
 [build-dependencies]
 cc = "1.0.83"
 
-[target.'cfg(not(target_os = "windows"))'.build-dependencies]
```
Comment (stormshield-kg) [inline pq-src/Cargo.toml:33]: This dependency is not necessary, it is enabled with the `vendored` feature of `openssl-sys`.

Comment (stormshield-kg) [reply pq-src/Cargo.toml:33]: The `DEP_OPENSSL_INCLUDE` variable is set by this line, defined in the build script of `openssl-sys`: https://github.com/sfackler/rust-openssl/blob/83b3186f5f0c82c28339c2b235b3bb4e5d324b7b/openssl-sys/build/main.rs#L108, so it is not required to add `openssl-src` as a dependency.

See the `DEP_<name>_<key>` section at https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts.

Comment (weiznich) [reply pq-src/Cargo.toml:33]: Thanks for clarifying, I expected that to be set by `openssl-src`, but having it set in `openssl-sys` simplifies things obviously.

File: Cargo.toml [hunk 2]
```diff
@@ -25,5 +25,6 @@ vcpkg = "0.2.6"
 
 [features]
 default = []
-bundled = ["pq-src"] 
+bundled = ["pq-src"]
+bundled-openssl = ["pq-src?/openssl-vendored"]
```
Comment (weiznich) [inline Cargo.toml:29]:
```suggestion
bundled-openssl = ["bundled", "pq-src?/openssl-vendored"]
```

File: Cargo.toml [hunk 3]
```diff
@@ -25,5 +25,6 @@ vcpkg = "0.2.6"
 
 [features]
 default = []
-bundled = ["pq-src"] 
+bundled = ["pq-src"]
+bundled-openssl = ["bundled", "pq-src?/openssl-vendored"]
```
[the remark this answers is not in the document]
Comment (stormshield-kg) [reply Cargo.toml]: Done.

**Review by weiznich — APPROVED** [review]

File: Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (weiznich) [reply Cargo.toml]: Thanks

**Pull request merged**